databases/jdp/database.go

```diff
@@ -86,9 +88,16 @@ func NewDatabase() (databases.Database, error) {
 		//Client:          databases.SecureHttpClient(),
 		Secret:          secret,
 		StagingRequests: make(map[uuid.UUID]StagingRequest),
+		DeleteAfter:     time.Duration(conf.Service.DeleteAfter) * time.Second,
 	}, nil
 }
 
+func DatabaseConstructor(conf config.Config) func() (databases.Database, error) {
```

Comment:
You know, this has an identical signature to `NewDatabase` above. Is there a reason not to just use `jdp.NewDatabase` in place of `DatabaseConstructor(conf)`? Go functions are closures so this syntax (note the parens) just works.

Reply:
I'm not quite following. The `DatabaseConstructor` function is used here:
https://github.com/mattldawson/dts/blob/11cf9ecc293a523ec4da1c5f076f17e6b10f6cd5/tasks/tasks.go#L91-L93

How would it look with using the `NewDatabase()` function?